File: Examples/android/CMeSdkJava/app_rework/src/main/java/me/connect/sdk/java/sample/backups/BackupsFragment.java
package me.connect.sdk.java.sample.backups;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import me.connect.sdk.java.sample.databinding.BackupsFragmentBinding;

public class BackupsFragment extends Fragment {

    private BackupsFragmentBinding binding;
    private BackupsViewModel model;

    public static BackupsFragment newInstance() {
        return new BackupsFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = BackupsFragmentBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        model = new ViewModelProvider(requireActivity()).get(BackupsViewModel.class);

        model.getLastBackup().observeOnce(getViewLifecycleOwner(), res -> {
            String text;
            if (res != null) {
                text = "Last backup: " + res;
            } else {
                text = "No backups yet";
            }
            binding.status.setText(text);
        });

        binding.buttonCreate.setOnClickListener(v -> {
            model.performBackup().observeOnce(getViewLifecycleOwner(), res -> {
                binding.status.setText(res);
            });
        });

        binding.buttonRestore.setOnClickListener(v -> {
            model.performRestore().observeOnce(getViewLifecycleOwner(), res -> {
                binding.status.setText(res);
            });
        });
    }
}

File: examples/android/CMeSdkJava/app_kt/src/main/kotlin/msdk/java/samplekt/backups/BackupsFragment.kt
package msdk.java.samplekt.backups

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import msdk.java.samplekt.databinding.BackupsFragmentBinding


class BackupsFragment : Fragment() {
    private lateinit var binding: BackupsFragmentBinding
    private val model: BackupsViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = BackupsFragmentBinding.inflate(inflater, container, false)
        return binding.getRoot()
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        model.getLastBackup().observeOnce(viewLifecycleOwner, Observer {
            val text = if (it != null) {
                "Last backup: $it"
            } else {
                "No backups yet"
            }
            binding.status.text = text
        })
        binding.buttonCreate.setOnClickListener {
            val backupKey = binding.backupKey.text.toString()
            model.performBackup(backupKey).observeOnce(viewLifecycleOwner, Observer { binding.status.text = it })
        }
        binding.buttonRestore.setOnClickListener {
            val backupKey = binding.backupKey.text.toString()
            model.performRestore(backupKey).observeOnce(viewLifecycleOwner, Observer { binding.status.text = it })
        }
    }

    companion object {
        fun newInstance(): BackupsFragment {
            return BackupsFragment()
        }
    }
}